micrometer-core/src/test/java/io/micrometer/core/instrument/binder/netty4/NettyEventExecutorMetricsTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.netty4;
+
+import io.micrometer.core.instrument.MockClock;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.simple.SimpleConfig;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.util.concurrent.SingleThreadEventExecutor;
+import org.junit.jupiter.api.Test;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link NettyEventExecutorMetrics}.
+ *
+ * @author Brian Clozel
+ */
+class NettyEventExecutorMetricsTests {
+
+    private SimpleMeterRegistry registry = new SimpleMeterRegistry(SimpleConfig.DEFAULT, new MockClock());
+
+    @Test
+    void shouldHaveTaskPendingMetric() throws Exception {
+        Set<String> names = new LinkedHashSet<>();
+        DefaultEventLoopGroup eventExecutors = new DefaultEventLoopGroup();
+        eventExecutors.spliterator().forEachRemaining(eventExecutor -> {
+            if (eventExecutor instanceof SingleThreadEventExecutor) {
+                SingleThreadEventExecutor singleThreadEventExecutor = (SingleThreadEventExecutor) eventExecutor;
+                names.add(singleThreadEventExecutor.threadProperties().name());
+                new NettyEventExecutorMetrics(eventExecutor).bindTo(this.registry);
```

Comment:
Without being a Netty expert, having users loop like this to bind each feels a bit weird to me. Would there be a reason a user would want metrics for some event executors in an EventExecutorGroup but not others? I wonder if we should use a higher level abstraction in NettyEventExecutorMetrics and add metrics for each executor for users rather than make them bind each one individually.

Reply:
I don't get this question.
Basically you have `EventLoopGroup` with `EventLoop`s. Every `EventLoop` has a name and a queue with pending tasks. What you are proposing is to have metrics on the `EventLoopGroup` is that correct?

Reply:
I'm saying we should have metrics on all of the EventLoops like this test does, but without making users do `new NettyEventExecutorMetrics(eventExecutor).bindTo(this.registry)` for each individual `EventExecutor`. Instead we could take the `EventLoopGroup` as a parameter and register metrics for each `EventExecutor` so users only need to call, e.g. `new NettyEventExecutorMetrics(eventLoopGroup).bindTo(this.registry)` once rather than iterating over each element like now. Basically the question is does it make sense to make things as granular as they are now? It only makes sense to me if there is a case you would only want metrics for some EventLoops in a group but not all of them. If you always want all of them, we should just take the group as a parameter and iterate internally so users don't have to. Does that make more sense?

Reply:
I agree with you, it is just easier in Reactor Netty at this point to do it per `EventLoop`. Definitely you want metrics for all `EventLoop`s in the `EventLoopGroup`.

Reply:
I guess I could add a constructor variant that takes the entire `EventLoopGroup`?
This implementation was targeting the "lazy" case where the current `EventLoop` is found in the given channel during its initialization. Something like:

```
@Override
public void initChannel(SocketChannel channel) throws Exception {
  ByteBufAllocator alloc = channel.alloc();
  if (alloc instanceof ByteBufAllocatorMetricProvider) {
    // this concurrent check must be implemented by micrometer users
    if (isAllocatorInstrumented(alloc)) {
      new AllocatorMetrics(((ByteBufAllocatorMetricProvider)alloc)).bindTo(prometheusRegistry);
    }
  }
  // this concurrent check must be implemented by micrometer users
  if (isEventLoopInstrumented(channel.eventLoop())) {
    new EventExecutorMetrics(channel.eventLoop()).bindTo(prometheusRegistry);
  }
  channel.pipeline().addLast(new HttpRequestDecoder());
  channel.pipeline().addLast(new HttpResponseEncoder());
  channel.pipeline().addLast(new CustomHttpServerHandler());
}
```